src/rulesets/Base/fastmath_able.jl

```diff
@@ -36,10 +36,86 @@ let
 
 
         # Unary complex functions
-        @scalar_rule abs(x::Real) sign(x)
-        @scalar_rule abs2(x) 2x
-        @scalar_rule angle(x::Real) Zero()
-        @scalar_rule conj(x::Real) One()
+        ## abs
+        function frule((_, Δx), ::typeof(abs), x::Real)
```

Comment:
Why are `frule`s and `rrule`s for `Real` inputs necessary? Doesn't the `@scalar_rule` macro work in those cases?

Reply:
Because `@scalar_rule` is still wrong for `x::Real` and `Δx::Complex`
```julia
using ChainRules
@scalar_rule abs(x::Real) sign(x)

jacobian_via_fdm(abs, 2.0)

#+RESULTS:
: 2×2 Array{Float64,2}:
:  1.0  -1.11022e-15
:  0.0   0.0
```
```julia
jacobian_via_frule(abs, 2.0)

#+RESULTS:
: 2×2 Array{Float64,2}:
:  1.0  0.0
:  0.0  1.0

```

Reply:
Good point!